### ISSUE TITLE
Brave news feed is offline even when reconnected to the internet

### ISSUE THREAD

**Issue by btlechowski** [brave/brave-browser#35361]
It is offline till browser restart

## Steps to Reproduce

1. Clean profile
2. Turn off the internet
3. Enable Brave News
4. Reconnect to the internet
5. Click `Refresh`

## Actual result:

Brave News shows `You appear to be offline`

![image](https://github.com/brave/brave-browser/assets/34715963/af41f66f-ba48-4bf6-b3a5-df21bf2df5ed)

## Expected result:
Brave news feed is loaded

## Reproduces how often: 

Easily reproduced

## Brave version (brave://version info)

Brave | 1.62.144 Chromium: 120.0.6099.217 (Official Build) beta (64-bit)
-- | --
Revision | 75de8d953f6a5686e425d46d7f0dc9284dc1c064
OS | Linux

cc @fallaciousreasoning @LorenzoMinto @rebron @LaurenWags

**Comment by MadhaviSeelam** [comment]
Reproduced the issue in Win 11 x64.  

example | example 
---- | -----
<img width="947" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/03e970d6-a455-424b-88f3-11c1227e360c">|<img width="947" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/ab6100c3-7616-4971-8283-b8ae76342ca8">

**Comment by LaurenWags** [comment]
Reproduced on macOS with STR and 

```
Brave | 1.62.147 Chromium: 120.0.6099.234 (Official Build) (x86_64)
-- | --
Revision | 3b25c3743150a54485dea24f0ceb1e69d6db51bc
OS | macOS Version 13.6.3 (Build 22G436)
```

**Comment by fallaciousreasoning** [comment]
Just double checking, but refreshing again fixes things? Or does it stay stuck?

**Comment by LorenzoMinto** [comment]
@LaurenWags @MadhaviSeelam is the feed stuck permanently or does it recover after refreshing?

**Comment by MadhaviSeelam** [comment]
@LorenzoMinto @fallaciousreasoning - refreshing adds `Add content sources` button but didn't refresh the feed.  I had to either subscribe or restart the browser.

example | example | example 
---- | ---- | ----
![image](https://github.com/brave/brave-browser/assets/98358127/f0fe21f4-9b75-4d07-b70c-b9b6280cb002)|![image](https://github.com/brave/brave-browser/assets/98358127/02430176-b6b5-4fd6-a5df-df09fdbd65d7)|![image](https://github.com/brave/brave-browser/assets/98358127/bad24ddd-a2b9-40f8-b8d3-e6f51498c78c)

**Comment by LorenzoMinto** [comment]
I'm also seeing this. Just to double check, this only happens when turning on Brave News the first time? @fallaciousreasoning might be something to do with re-downloading `sources.json` files?

**Comment by fallaciousreasoning** [comment]
Yup, should only happen if you enable Brave News while you're offline - I think it should be the same in the old Brave News too
